python/tests/data_structure/test_limbs.py

```diff
@@ -0,0 +1,152 @@
+    assert limbs.get_parts() is None
+    assert limbs.get_points() is None
+    # test conn + parts
+    parts = [[0, 1, 2, 3, 4], [5, 6, 7, 8, 9]]
+    limbs = Limbs(connections=connections, parts=parts)
+    assert limbs.get_connections().shape == (10, 2)
+    assert len(limbs.get_parts()) == 2
+    assert limbs.get_points() is None
+    # test conn + parts + points
+    points = np.zeros(shape=[20, 3])
+    limbs = Limbs(connections=connections, parts=parts, points=points)
+    assert limbs.get_connections().shape == (10, 2)
+    assert len(limbs.get_parts()) == 2
+    assert len(limbs.get_points()) == 20
+    # test torch
+    points = np.zeros(shape=[20, 3])
+    limbs = Limbs(
+        connections=torch.from_numpy(connections),
+        parts=parts,
+        points=torch.from_numpy(points))
+    assert isinstance(limbs.get_connections(), np.ndarray)
+    assert limbs.get_connections().shape == (10, 2)
+    assert len(limbs.get_parts()) == 2
+    assert len(limbs.get_points()) == 20
+
+
+def test_set_conections():
+    connections = np.zeros(shape=[10, 2])
+    limbs = Limbs(connections=connections)
+    # set np
+    limbs.set_connections(connections + 1)
+    assert limbs.get_connections()[0, 0] == 1
+    # set torch
+    limbs.set_connections(torch.from_numpy(connections + 2))
+    assert limbs.get_connections()[0, 0] == 2
+    # set list
+    with pytest.raises(TypeError):
+        limbs.set_connections(connections.tolist())
+    # set wrong shape
+    with pytest.raises(ValueError):
+        limbs.set_connections(connections.reshape(2, 5, 2))
+    # set names
+    connection_names = [str(x) for x in range(len(connections))]
+    limbs.set_connections(connections, connection_names)
+    assert len(limbs.connection_names) == len(connection_names)
+    # set wrong len, warning
+    limbs.set_connections(connections, connection_names[:5])
+    assert limbs.connection_names is None
+
+
+def test_set_parts():
+    connections = np.zeros(shape=[10, 2])
+    limbs = Limbs(connections=connections)
+    parts = [[0, 1, 2, 3, 4], [5, 6, 7, 8, 9]]
+    # set parts
+    limbs.set_parts(parts)
+    # set wrong part type
+    with pytest.raises(TypeError):
+        limbs.set_parts(np.asarray(parts))
+    # set wrong index type
+    with pytest.raises(TypeError):
+        limbs.set_parts(np.asarray(parts).astype(np.float64).tolist())
+    # set names
+    part_names = [str(x) for x in range(len(parts))]
+    limbs.set_parts(parts, part_names)
+    assert len(limbs.part_names) == len(part_names)
+    # set wrong len, warning
+    limbs.set_parts(parts, part_names[:1])
+    assert limbs.part_names is None
+
+
+def test_set_points():
+    connections = np.zeros(shape=[10, 2])
+    limbs = Limbs(connections=connections)
+    # test points2d
+    points = np.zeros(shape=[20, 2])
+    limbs.set_points(points)
+    # test points3d
+    points = np.zeros(shape=[20, 3])
+    limbs.set_points(points)
+    # test points3d+conf
```

Comment:
How to represent points2d+conf?

Reply:
`test points3d` -> `test points3d/points2d_conf`. Fixed in 
593876b.